src/umpire/Umpire.cpp

```diff
@@ -27,21 +27,23 @@ void print_allocator_records(Allocator allocator, std::ostream& os) {
 
   auto strategy = allocator.getAllocationStrategy();
 
-  rm.m_allocations.print([strategy] (const util::AllocationRecord* rec) {
-    return rec->m_strategy == strategy;
+  rm.m_allocations.print([strategy] (const util::AllocationRecord& rec) {
+    return rec.m_strategy == strategy;
   }, os);
 }
 
-std::vector<const util::AllocationRecord*> get_allocator_records(Allocator allocator) {
-  auto& rm = umpire::ResourceManager::getInstance();
+std::vector<const util::AllocationRecord*> get_allocator_records(Allocator UMPIRE_UNUSED_ARG(allocator)) {
+  // auto& rm = umpire::ResourceManager::getInstance();
 
-  auto strategy = allocator.getAllocationStrategy();
+  // auto strategy = allocator.getAllocationStrategy();
+
+  // TODO TBD
 
   std::vector<const util::AllocationRecord*> recs;
-  std::copy_if(rm.m_allocations.begin(), rm.m_allocations.end(),
-               std::back_inserter(recs), [strategy] (const util::AllocationRecord* rec) {
-                 return rec->m_strategy == strategy;
-               });
+  // std::copy_if(rm.m_allocations.begin(), rm.m_allocations.end(),
+  //              std::back_inserter(recs), [strategy] (const util::AllocationRecord& rec) {
+  //                return rec.m_strategy == strategy;
+  //              });
```

Comment:
I don't understand any of the commented out code above and the 'TODO TBD' is nebulous.  Please describe and/or remove.

Reply:
This method returns a vector of allocation records created by a specific `Allocator`. I created it a while back because it was useful for inspecting mixed pools. When I changed the `ResourceManager` API, I quickly commented out this method, but yes, it will be updated before merging.